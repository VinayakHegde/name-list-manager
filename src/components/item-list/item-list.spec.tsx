import * as React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import ItemList from ".";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const withProps = (props) => {
  return <ItemList {...props} />;
};

describe("<DisplayItem />", () => {
  const props = {
    list: ["Hello", "word", "world"],
    remove: jest.fn(),
  };
  it("should match the snapshot", async () => {
    act(() => {
      render(withProps(props), container);
    });
    expect(container).toMatchSnapshot();
  });
  it("should render elements", async () => {
    act(() => {
      render(withProps(props), container);
    });
    const displayItem = container.querySelectorAll(".displayItem");
    expect(displayItem).toHaveLength(props.list.length);
  });
});
