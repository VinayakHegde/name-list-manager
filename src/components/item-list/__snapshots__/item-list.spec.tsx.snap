// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayItem /> should match the snapshot 1`] = `
<div>
  <section
    class="itemList"
  >
    <div
      class="displayItem"
    >
      <span
        class="displayItem__name"
      >
        Hello
      </span>
      <button
        class="danger"
      >
        Remove
      </button>
    </div>
    <div
      class="displayItem"
    >
      <span
        class="displayItem__name"
      >
        word
      </span>
      <button
        class="danger"
      >
        Remove
      </button>
    </div>
    <div
      class="displayItem"
    >
      <span
        class="displayItem__name"
      >
        world
      </span>
      <button
        class="danger"
      >
        Remove
      </button>
    </div>
  </section>
</div>
`;
